packages/amplify-graphql-api-construct-tests/src/__tests__/owner-auth/assoc-field/assoc-field.test.ts

```diff
@@ -0,0 +1,720 @@
+      });
+
+      test('getPrimary shows relations if created by same owner', async () => {
+        await testGetPrimaryVisibleForSameOwner(currentId, apiEndpoint, accessToken);
+      });
+
+      test('getPrimary redacts relations if created by same owner', async () => {
+        await testGetPrimaryRedactedForDifferentOwners(currentId, apiEndpoint, accessToken, accessToken2);
+      });
+    });
+
+    describe('RelatedOne as source model', () => {
+      test('createRelatedOne shows relations if created by same owner', async () => {
+        await testCreateRelatedOneVisibleForSameOwner(currentId, apiEndpoint, accessToken);
+      });
+
+      test('createRelatedOne redacts relations if created by same owner', async () => {
+        await testCreateRelatedOneRedactedForDifferentOwners(currentId, apiEndpoint, accessToken, accessToken2);
+      });
+
+      test('updateRelatedOne shows relations if created by same owner', async () => {
+        await testUpdateRelatedOneVisibleForSameOwner(currentId, apiEndpoint, accessToken);
+      });
+
+      test('updateRelatedOne redacts relations if created by different owner', async () => {
+        await testUpdateRelatedOneRedactedForDifferentOwners(currentId, apiEndpoint, accessToken, accessToken2);
+      });
+
+      test('getRelatedOne shows relations if created by same owner', async () => {
+        await testGetRelatedOneVisibleForSameOwner(currentId, apiEndpoint, accessToken);
+      });
+
+      test('getRelatedOne redacts relations if created by same owner', async () => {
+        await testGetRelatedOneRedactedForDifferentOwners(currentId, apiEndpoint, accessToken, accessToken2);
+      });
+    });
+
+    describe('RelatedMany as source model', () => {
+      test('createRelatedMany shows relations if created by same owner', async () => {
+        await testCreateRelatedManyVisibleForSameOwner(currentId, apiEndpoint, accessToken);
+      });
+
+      test('createRelatedMany redacts relations if created by different owner', async () => {
+        await testCreateRelatedManyRedactedForDifferentOwners(currentId, apiEndpoint, accessToken, accessToken2);
+      });
+
+      test('updateRelatedMany shows relations if created by same owner', async () => {
+        await testUpdateRelatedManyVisibleForSameOwner(currentId, apiEndpoint, accessToken);
+      });
+
+      test('updateRelatedMany redacts relations if created by different owner', async () => {
+        await testUpdateRelatedManyRedactedForDifferentOwners(currentId, apiEndpoint, accessToken, accessToken2);
+      });
+
+      test('getRelatedMany shows relations if created by same owner', async () => {
+        await testGetRelatedManyVisibleForSameOwner(currentId, apiEndpoint, accessToken);
+      });
+
+      test('getRelatedMany redacts relations if created by same owner', async () => {
+        await testGetRelatedManyRedactedForDifferentOwners(currentId, apiEndpoint, accessToken, accessToken2);
+      });
+    });
+  });
+});
```

Comment:
There's a lot of tests including different data source types in this suite. Wondering if it hits the token expiry limit or was it reliably succeeding on CI?

Reply:
That's a very good question. It succeeds reliably on my local machine, but I haven't yet run it in CI. I probably need to make sure this suite is run solo though

Reply:
If we hit token expiry, IN an upcoming PR, I'll split each of the "source model" variations into their own test file so they can be run independently.